pyproject.toml

```diff
@@ -9,7 +9,6 @@ dependencies = []
 [tool.uv]
 dev-dependencies = [
     "arro3-core>=0.4.2",
-    "black>=24.10.0",
```

Comment:
This is used in the docs to format code docstrings https://mkdocstrings.github.io/python/usage/#options-summary. It looks like the most recent version of mkdocstrings added the option for that to be done using ruff as well as black, so we can remove black as long as we ensure we're using the latest mkdocstrings.

Reply:
Gotcha, will double check

Reply:
@kylebarron done. I lower bound `mkdocstrings` and added `ruff` as dev dependency instead. now both the docs and the source code are using `ruff` as formatter